.github/files/update-project-mirrors.sh

```diff
@@ -0,0 +1,154 @@
+#!/bin/bash
+
+git_setup()
+{
+  cat <<- EOF > "$HOME/.netrc"
+		machine github.com
+			login matticbot
+			password $GITHUB_TOKEN
+		machine api.github.com
+			login matticbot
+			password $GITHUB_TOKEN
+EOF
+  chmod 600 "$HOME/.netrc"
+
+  git config --global user.email "matticbot@users.noreply.github.com"
+  git config --global user.name "matticbot"
+}
+
+# Halt on error
+set -eo pipefail
+
+if [[ -n "$CI" ]]; then
+	git_setup
+fi
+
+BASE=$(pwd)
+CLONE_BASE=$(mktemp -d "${TMPDIR:-/tmp}/jetpack-update-project-mirrors.XXXXXXXX")
+MONOREPO_COMMIT_MESSAGE=$(git show -s --format=%B $GITHUB_SHA)
+COMMIT_MESSAGE=$( echo "${MONOREPO_COMMIT_MESSAGE}\n\nCommitted via a GitHub action: https://github.com/automattic/jetpack/runs/${GITHUB_RUN_ID}" )
+COMMIT_ORIGINAL_AUTHOR="${GITHUB_ACTOR} <${GITHUB_ACTOR}@users.noreply.github.com>"
+
+if [[ "$GITHUB_REF" =~ ^refs/heads/ ]]; then
+	BRANCH=${GITHUB_REF#refs/heads/}
+else
+	echo "Could not determine branch name from $GITHUB_REF"
+	exit 1
+fi
+
+# Install Yarn generally.
+echo "::group::Monorepo setup"
+yarn install
+echo "::endgroup::"
+
+echo "Cloning projects and pushing to Automattic mirror repos"
+EXIT=0
+
+# sync to read-only clones
+for project in projects/packages/* projects/plugins/*; do
+	PROJECT_DIR="${BASE}/${project}"
+	[[ -d "$PROJECT_DIR" ]] || continue # We are only interested in directories (i.e. projects)
+
+	printf "\n\n\e[7m Project: %s \e[0m\n" "$project"
+
+	cd "${PROJECT_DIR}"
+
+	# Read mirror repo from composer.json, if it exists
+	if [[ -f "composer.json" ]]; then
+		COMPOSER_JSON_EXISTED=true
+		GIT_SLUG=$(jq -r '.extra["mirror-repo"] // ( .name | sub( "^automattic/"; "Automattic/" ) )' composer.json)
+		if [[ -z "$GIT_SLUG" ]]; then
+			echo "::error::Failed to determine mirror repo from composer.json"
+			EXIT=1
+			continue
+		fi
+	else
+		COMPOSER_JSON_EXISTED=false
+		GIT_SLUG="Automattic/jetpack-${project##*/}"
```

Comment:
What do you think about being opinionated and stating that all projects should have a composer.json file?

Reply:
I did that at first, then found that projects/packages/debug-helper currently doesn't.

I wouldn't object to a followup adding one and removing this fallback, or moving it elsewhere if it's not really a package.

Reply:
#18317 handles moving it _and_ adding a composer.json :-D